packages/perseus-core/src/utils/get-save-warnings-for-item.ts

```diff
@@ -0,0 +1,74 @@
+import type {PerseusItem, PerseusWidgetsMap, RadioWidget} from "../data-schema";
+
+/* ------- Main save warnings function for the whole item ------- */
+
+// TODO(LEMS-3643): Remove the "WORK IN PROGRESS" comment below once all
+// widgets have been migrated and the function is ready for use.
+/**
+ * WORK IN PROGRESS: Get the save warnings (blocking issues) for a given PerseusItem.
+ *
+ * @param item - The PerseusItem that will be fully parsed and checked
+ *                for blocking changes that should prevent the item from being
+ *                saved/published.
+ * @returns An array of strings, each representing a warning that should prevent
+ *          the item from being saved/published.
+ */
+export function getSaveWarningsForItem(item: PerseusItem): Array<string> {
+    const allSaveWarnings: Array<string> = [];
+    const widgets: PerseusWidgetsMap = item.question.widgets;
+    const content = item.question.content;
+
+    // Only check widgets that are actually referenced in the content
+    // (widgets can remain in the widgets object after being deleted from content)
+    for (const [widgetId, widget] of Object.entries(widgets)) {
```

Comment:
I think we should use the `traverse()` helper function to go through an item, because we could have complex widgets (like `graded-group`) which contain other widgets.  I _think_ we want to validate widgets inside those also, right? 

See: https://github.com/Khan/perseus/blob/ae58c67f9c07bf2251f183942c2466bb5ec05108/packages/perseus-core/src/traversal.ts#L93